scripts/xref_check.eterm

```diff
@@ -0,0 +1,536 @@
+           {keccakf1600_fips202,sha3_256,1}},
+          {{jose_sha3_keccakf1600_driver,sha3_384,1},
+           {keccakf1600_fips202,sha3_384,1}},
+          {{jose_sha3_keccakf1600_driver,sha3_512,1},
+           {keccakf1600_fips202,sha3_512,1}},
+          {{jose_sha3_keccakf1600_driver,shake128,2},
+           {keccakf1600_fips202,shake128,2}},
+          {{jose_sha3_keccakf1600_driver,shake256,2},
+           {keccakf1600_fips202,shake256,2}},
+          {{jose_sha3_keccakf1600_nif,sha3_224,1},{keccakf1600,hash,2}},
+          {{jose_sha3_keccakf1600_nif,sha3_256,1},{keccakf1600,hash,2}},
+          {{jose_sha3_keccakf1600_nif,sha3_384,1},{keccakf1600,hash,2}},
+          {{jose_sha3_keccakf1600_nif,sha3_512,1},{keccakf1600,hash,2}},
+          {{jose_sha3_keccakf1600_nif,shake128,2},{keccakf1600,hash,3}},
+          {{jose_sha3_keccakf1600_nif,shake256,2},{keccakf1600,hash,3}},
+          {{jose_sha3_libdecaf,sha3_224,1},{libdecaf_sha3,hash,2}},
+          {{jose_sha3_libdecaf,sha3_256,1},{libdecaf_sha3,hash,2}},
+          {{jose_sha3_libdecaf,sha3_384,1},{libdecaf_sha3,hash,2}},
+          {{jose_sha3_libdecaf,sha3_512,1},{libdecaf_sha3,hash,2}},
+          {{jose_sha3_libdecaf,shake128,2},{libdecaf_sha3,hash,3}},
+          {{jose_sha3_libdecaf,shake256,2},{libdecaf_sha3,hash,3}},
+          {{lwm2m_coap_client,channel_apply,3},{lwm2m_coap_dtls_socket,close,1}},
+          {{lwm2m_coap_client,channel_apply,3},{lwm2m_coap_dtls_socket,connect,2}},
+          {{lwm2m_coap_client,channel_apply,3},{lwm2m_coap_udp_socket,close,1}},
+          {{lwm2m_coap_client,channel_apply,3},
+           {lwm2m_coap_udp_socket,get_channel,2}},
+          {{lwm2m_coap_client,channel_apply,3},{lwm2m_coap_udp_socket,start_link,0}},
+          {{lwm2m_coap_client,test,0},{eunit,test,1}},
+          {{lwm2m_coap_message_parser,test,0},{eunit,test,1}},
+          {{lwm2m_coap_observer,init,1},{lwm2m_coap_dtls_socket,connect,2}},
+          {{lwm2m_coap_observer,init,1},{lwm2m_coap_udp_socket,get_channel,2}},
+          {{lwm2m_coap_observer,init,1},{lwm2m_coap_udp_socket,start_link,0}},
+          {{lwm2m_coap_observer,terminate,2},{lwm2m_coap_dtls_socket,close,1}},
+          {{lwm2m_coap_observer,terminate,2},{lwm2m_coap_udp_socket,close,1}},
+          {{ranch_app,consider_profiling,0},{eprof,start,0}},
+          {{ranch_app,consider_profiling,0},{eprof,start_profiling,1}},
+          {{ranch_app,profile_output,0},{eprof,analyze,1}},
+          {{ranch_app,profile_output,0},{eprof,log,1}},
+          {{ranch_app,profile_output,0},{eprof,stop_profiling,0}}
+         ]
+   }
+
+  %% Check undefined_functions
+, #{ name => undefined_functions
+   , analysis => undefined_functions
+   , excl_apps =>
+         [ observer
+         ]
+   , excl_mods =>
+         [ systools
+         , systools_make
+         , release_handler
+         , systools_relup
+         , cuttlefish_rebar_plugin
+         ]
+   , filters =>
+         [{'Elixir.Atom',to_string,1},
```

Comment:
Those functions are absent only on non-Elixir releases, right?  Maybe they should be checked for in the Elixir release?

Reply:
yes. should be part of the CI in Elixir build I guess. Is there a flag to check if it is Elixir runtime?

Reply:
There's a variable that's set in `emqx_vars` (`IS_ELIXIR` I think), but that should appear only in 5.0 :thinking: